packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -302,20 +333,31 @@ export class ComputeEnvironment extends Resource implements IComputeEnvironment
     const spotFleetRole = this.getSpotFleetRole(props);
     let computeResources: CfnComputeEnvironment.ComputeResourcesProperty | undefined;
 
-    // Only allow compute resources to be set when using UNMANAGED type
-    if (props.computeResources && !this.isManaged(props)) {
+    // Only allow compute resources to be set when using MANAGED type
+    if (props.computeResources && this.isManaged(props)) {
       computeResources = {
-        allocationStrategy: props.allocationStrategy || AllocationStrategy.BEST_FIT,
+        allocationStrategy: props.computeResources.allocationStrategy
+         || (
+             props.computeResources.type === ComputeResourceType.SPOT
+             ? AllocationStrategy.SPOT_CAPACITY_OPTIMIZED
+             : AllocationStrategy.BEST_FIT
+            ),
         bidPercentage: props.computeResources.bidPercentage,
         desiredvCpus: props.computeResources.desiredvCpus,
         ec2KeyPair: props.computeResources.ec2KeyPair,
         imageId: props.computeResources.image && props.computeResources.image.getImage(this).imageId,
         instanceRole: props.computeResources.instanceRole
-          ? props.computeResources.instanceRole.roleArn
-          : new iam.Role(this, 'Resource-Instance-Role', {
-            assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
-          }).roleArn,
+          ? props.computeResources.instanceRole
+          : new iam.CfnInstanceProfile(this, 'Instance-Profile', {
```

Comment:
Can you explain the reasoning behind this change?

Reply:
Related to https://github.com/aws/aws-cdk/issues/6495 i imagine right?

Reply:
Correct. I wrote a few words about it a couple of comments above.

Will add that and the related issue to the PR description.